### ISSUE TITLE
os_patching_fact_generation.ps1 created without required permissions

### ISSUE THREAD

**Issue by jscottPTC** [albatrossflavour/puppet_os_patching#206]
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 7.6.1
- Ruby on Server: 3.0.1p64
- Ruby on Client: 3.0.1p64
- Server Distribution: Ubuntu 20.04.2
- Client Distribution: Windows 10 (OS build 19042.928
- Client Powershell version: 5.1
- Module version: 0.16.0

## How to reproduce (e.g Puppet code you use)

1) On client Install 64 bit Puppet Agent for windows from msi: [link to msi I used](http://downloads.puppetlabs.com/windows/puppet/puppet-agent-7.6.1-x64.msi)

2) On the server install OS Patcher with the command sudo /opt/puppetlabs/bin/puppet module install albatrossflavour-os_patching --version
0.16.0
3) On the server create file called 'os_patcher.pp' with one line of text 'include os_patching'.  Place this file under '/code/environments/production/manifests/'

4) On client start command prompt with puppet.  Run it as administrator.
5) Register agent with puppetserver
6) On client run 'puppet agent -t'

## What are you seeing
When I run the command puppet agent -t there is an error in the output saying that it cannot read os_patching_fact_generation.ps1
## What behaviour did you expect instead
puppet agent -t runs without error
## Output log
[log.txt](https://github.com/albatrossflavour/puppet_os_patching/files/6540970/log.txt)

**Comment by fangludi** [comment]
I have the same issue.

**Comment by BKNR** [comment]
I have the same issue using Puppet 6.21.1 on the client.

**Comment by jscottPTC** [comment]
I'm not sure this is actually a problem.  It appears that the service runs under the Local System account.  The Local System Account seems to have more privileges.  I don't want to say it has _sufficient_ privileges because I am running into another issue that is preventing this from working that may be caused by permissions but at this point I am not 100 percent sure..

**Comment by SeanBatt** [comment]
I had issues running 'puppet agent -t' from an elevated powershell. I noticed the default permissions for the os_patching_fact_generation.ps1 were set to '0000' in `os_patching/data/kernel/windows.yaml` by `os_patching::fact_mode: '0000'` (an addition from the most recent commit, 554165b) which may have left it only usable by SYSTEM? I've changed the class call to include `class { 'os_patching':
    ...
    fact_mode        => '0770',
  }
` which may well have improved things for me, allowing the script to run when `puppet agent` is run.
